libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -237,6 +237,30 @@ string YulUtilFunctions::shiftRightFunction(size_t _numBits)
 	}
 }
 
+string YulUtilFunctions::updateByteSliceFunction(size_t _numBytes, size_t _shiftBytes)
+{
+	solAssert(_numBytes <= 32, "");
+	solAssert(_shiftBytes <= 32, "");
+	size_t numBits = _numBytes * 8;
+	size_t shiftBits = _shiftBytes * 8;
+	string functionName = "update_byte_slice_" + to_string(_numBytes) + "_shift_" + to_string(_shiftBytes);
+	return m_functionCollector->createFunction(functionName, [&]() {
+		return
+			Whiskers(R"(
+			function <functionName>(value, toInsert) -> result {
```

Comment:
I'm not sure I fully understand the logic in this function. Could you briefly clarify?

Reply:
The bits defined by `mask` are first set to zero in `value` and then set to the corresponding bits from `toInsert`